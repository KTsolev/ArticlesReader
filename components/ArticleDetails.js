
import React from "react";
import { View, Text, Button } from "react-native";
import { headerConfiguration } from './Header';
import { WebView } from 'react-native-webview';

class ArticleDetails extends React.Component {
    static navigationOptions = headerConfiguration;
    render() {
      const { navigation } = this.props;
      const itemId = navigation.getParam('itemId', 'NO-ID');
      const url = navigation.getParam('url', 'some default value');
      return (
        <WebView
          source={{ uri: url }}
          startInLoadingState={false}
          javaScriptEnabled={true}
          thirdPartyCookiesEnabled={true}
          scalesPageToFit
        />
      );
    }
  }
 
export default ArticleDetails;