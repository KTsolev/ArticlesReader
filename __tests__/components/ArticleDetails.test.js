import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

// Component to be tested
import ArticleDetails from '../../components/ArticleDetails/ArticleDetails';

const createTestProps = (props) => ({
    navigation: {
      navigate: jest.fn(),
      getParam: (name) => {
          if (name==='url') {
            return 'https://jestjs.io/';
          } 
          return null;
      }
    },
    ...props
  });

  describe('<ArticleDetails />', () => {
      let props;
      beforeEach(() => {
        props = createTestProps({});
      });
      test('renders the component', () => {
          const wrapper = shallow(<ArticleDetails {...props} />);
          const component = wrapper.dive();
          expect(toJson(component)).toMatchSnapshot();
      });
  });