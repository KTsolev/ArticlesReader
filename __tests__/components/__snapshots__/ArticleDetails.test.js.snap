// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleDetails /> renders the component 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "overflow": "hidden",
    }
  }
>
  <_class
    cacheEnabled={true}
    javaScriptEnabled={true}
    key="webViewKey"
    messagingEnabled={false}
    onLoadingError={[Function]}
    onLoadingFinish={[Function]}
    onLoadingProgress={[Function]}
    onLoadingStart={[Function]}
    onMessage={[Function]}
    onShouldStartLoadWithRequest={[Function]}
    scalesPageToFit={true}
    sharedCookiesEnabled={true}
    source={
      Object {
        "uri": "https://jestjs.io/?api-key=qWO5J56rRa4sdwmDlndp4wv5EEAZecAR",
      }
    }
    startInLoadingState={true}
    style={
      Array [
        Object {
          "flex": 1,
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "#ffffff",
        },
        undefined,
      ]
    }
    styles={
      Object {
        "position": "relative",
      }
    }
    thirdPartyCookiesEnabled={true}
    useSharedProcessPool={true}
  />
  <Loader />
</View>
`;
